src/app/shared/form/form.component.ts

```diff
@@ -0,0 +1,244 @@
+      this.type = data.type;
+      this.isCreate = (this.type === 'create'? true:false);
+      // Load metadata and form information
+      this.globalMetadata = this.metadataService.getInfoMetadata(formKind+'Info')[0];
+      this.formMetadata = this.metadataService.getFormMetadata(formKind);
+      this.title = this.globalMetadata['title'];
+      this.customform = this.globalMetadata['customform'];
+      titleService.set([this.title]);
+      // Load metadata and form information
+      if(this.type === 'edit'){
+        this.getIndex();
+        this.loadEdit(); // Load data for editing
+      }else{
+        this.isloaded = true
+      }
+    });
+    // Add this.mySubscription to UnsubscribeService
+    this.unsubscribeService.add(this.mySubscription);
+  }
+
+  /**
+   * Loads data for editing a form.
+  */
+  getIndex() {
+    this.route.params.subscribe((params: Params) => {
+      this.editedIndex = +params['id'];
+    });
+  }
+
+  /**
+   * Loads data for editing a form.
+   */
+  loadEdit() {
+    this.route.params.subscribe((params: Params) => {
+      this.editedIndex = +params['id'];
+    });
+
+    // Fetch data from the API for editing
+    this.gs.get(this.apiPath, this.editedIndex).subscribe((result) => {
+      this.formValues = result;
+      this.isloaded = true; // Data is loaded and ready for form rendering
+    });
+  }
+
+  /**
+   * Angular lifecycle hook: ngOnInit
+   */
+  ngOnInit(): void {
+    // If in "edit" mode, load data for editing
+    if (this.type === 'edit') {
+      this.loadEdit();
+    }
+  }
+
+  /**
+   * Angular lifecycle hook: ngOnDestroy
+   * Unsubscribe from all subscriptions to prevent memory leaks.
+   */
+  ngOnDestroy(): void {
+    this.unsubscribeService.unsubscribeAll();
+  }
+
+  /**
+   * Handles the submission of the form.
+   * @param formValues - The values submitted from the form.
+  */
+  onFormSubmit(formValues: any) {
+    if(this.customform){
+      this.onCustomForm(formValues);
+    }
+    if (this.type === 'create') {
+      // Create mode: Submit form data for creating a new item
+      console.log(formValues)
+      this.mySubscription = this.gs.create(this.apiPath, formValues).subscribe(() => {
+        this.alert.okAlert(this.globalMetadata['submitok'], '');
+        this.router.navigate([this.globalMetadata['submitokredirect']]);
+      });
+    } else {
+      // Update mode: Submit form data for updating an existing item
+      this.mySubscription = this.gs.update(this.apiPath, this.editedIndex, formValues).subscribe(() => {
+        this.alert.okAlert(this.globalMetadata['submitok'], '');
+        this.router.navigate([this.globalMetadata['submitokredirect']]);
+      });
+    }
+  }
+
+  /**
+   * Modifies the form values as needed for custom form submission.
+   * @param formValues - The form values to be modified.
+   * @returns The modified form values.
+   */
+  onCustomForm(formValues: any) {
```

Comment:
The name of this function is kind of vague. `modifyFormValues` or something is probably more descriptive?

Reply:
modifyFormValues makes more sense